File: app/src/main/java/io/github/romangolovanov/apps/ametro/ui/navigation/adapter/NavigationTextItemHolder.java
package io.github.romangolovanov.apps.ametro.ui.navigation.adapter;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import io.github.romangolovanov.apps.ametro.R;
import io.github.romangolovanov.apps.ametro.ui.navigation.entities.NavigationItem;
import io.github.romangolovanov.apps.ametro.ui.navigation.entities.NavigationTextItem;

class NavigationTextItemHolder implements IHolder
{
    private final ImageView imageView;
    private final TextView textView;
    private final View container;

    public NavigationTextItemHolder(View view){
        imageView = (ImageView)view.findViewById(R.id.icon);
        textView = (TextView)view.findViewById(R.id.text);
        this.container = view;
    }

    @Override
    public void update(NavigationItem item) {
        NavigationTextItem textItem = (NavigationTextItem)item;
        imageView.setImageDrawable(textItem.getDrawable());
        textView.setText(textItem.getText());
        container.setBackgroundResource(textItem.isSelected() ? R.color.activated_color : android.R.color.transparent);
    }
}

File: app/src/main/java/io/github/romangolovanov/apps/ametro/ui/navigation/adapter/NavigationTextItemHolder.kt
package io.github.romangolovanov.apps.ametro.ui.navigation.adapter

import android.view.View
import android.widget.ImageView
import android.widget.TextView
import io.github.romangolovanov.apps.ametro.R
import io.github.romangolovanov.apps.ametro.ui.navigation.entities.NavigationItem
import io.github.romangolovanov.apps.ametro.ui.navigation.entities.NavigationTextItem

internal class NavigationTextItemHolder(view: View) : IHolder {

    private val imageView: ImageView = view.findViewById(R.id.icon)
    private val textView: TextView = view.findViewById(R.id.text)
    private val container: View = view

    override fun update(item: NavigationItem) {
        val textItem = item as NavigationTextItem
        imageView.setImageDrawable(textItem.getDrawable())
        textView.text = textItem.text
        container.setBackgroundResource(if (textItem.selected) R.color.activated_color else android.R.color.transparent)
    }
}
